public/account_settings.php

```
@@ -60,4 +55,22 @@ function compare_passwords($p1, $p2)
 
 <?php } ?>
 
+<?php
+if (isset($_POST['submit'])) {
+
+  empty($_POST["first_name"]) ?: $user->first_name = $_POST["first_name"];
+  empty($_POST["last_name"]) ?: $user->last_name = $_POST["last_name"];
+  empty($_POST["student_id"]) ?: $user->student_id = $_POST["student_id"];
+  empty($_POST["email"]) ?: $user->email = $_POST["email"];
```

😂 man this is so not DRY

---

btw this form update is going to fail silently if anything goes wrong, e.g. if we aren't connected to the database, or if the student iD is not an int

---

thats fair, ill add the type checking where it should be

---

Ok I think the student id validation should be good, made a few more changes, and a database error shouldnt be silent anymore, let me know if i missed something else